components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,337 @@
+        UIContentSizeCategoryAccessibilityExtraLarge : @34,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @38,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @42
+      };
+      _scalingCurve = scalingCurve;
+    }
+  }
+
+  return self;
+}
+
+- (UIFont *)scalableFontWithFont:(UIFont *)font {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    sizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  UIFont *scaledFont;
+  if (font.mdc_scalingCurve) {
+    scaledFont = [font mdc_scaledFontForSizeCategory:sizeCategory];
+    scaledFont.mdc_scalingCurve = _scalingCurve;
+  } else {
+    // The original font does not have a scaling curve, so create a copy some we aren't
+    // attaching associated objects to system fonts
+    scaledFont = [font copy];
+    scaledFont.mdc_scalingCurve = _scalingCurve;
+    scaledFont = [scaledFont mdc_scaledFontForSizeCategory:sizeCategory];
+  }
+
+  return scaledFont;
+}
+
+- (CGFloat)scaledValueForValue:(CGFloat)value {
+  // If it is available, query the preferredContentSizeCategory.
+  UIContentSizeCategory defaultSizeCategory = UIContentSizeCategoryLarge;
+  UIContentSizeCategory currentSizeCategory = UIContentSizeCategoryLarge;
+  if ([UIApplication mdc_safeSharedApplication]) {
+    currentSizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+  } else if (@available(iOS 10.0, *)) {
+    currentSizeCategory = UIScreen.mainScreen.traitCollection.preferredContentSizeCategory;
+  }
+
+  NSNumber *defaultFontSizeNumber = _scalingCurve[defaultSizeCategory];
+
+  NSNumber *currentFontSizeNumber = _scalingCurve[currentSizeCategory];
+  // If you have queried the table for a sizeCategory that doesn't exist, we will return the
+  // traits for XXXL.  This handles the case where the values are requested for one of the
+  // accessibility size categories beyond XXXL such as
+  // UIContentSizeCategoryAccessibilityExtraLarge.
+  if (currentFontSizeNumber == nil) {
+    currentFontSizeNumber = _scalingCurve[UIContentSizeCategoryExtraExtraExtraLarge];
+  }
+
+  // Guard against broken / incomplete scaling curves by returning self if fontSizeNumber is nil.
+  if (currentFontSizeNumber == nil || defaultFontSizeNumber == nil) {
+    return value;
+  }
+
+  CGFloat currentFontSize = (CGFloat)currentFontSizeNumber.doubleValue;
+  CGFloat defaultFontSize = (CGFloat)currentFontSizeNumber.doubleValue;
+
+  // Guard against broken / incomplete scaling curves by returning self if fontSize <= 0.0.
+  if (currentFontSize <= 0.0 || defaultFontSize <= 0.0) {
+    return value;
+  }
+
+  return currentFontSize / defaultFontSize;
```

Comment:
Shouldn't `value` be used in this computation somehow? I would expect this ratio to be applied to `value` as a multiplier. For example

```objc
value = 8;
defaultFontSize = 10;
currentFontSize = 15;
returnedValue = (15 / 10) * 8 = 12;
```

Reply:
Fixed.